language_service/src/completion.rs

```diff
@@ -19,11 +19,49 @@ const PRELUDE: [&str; 3] = [
     "Microsoft.Quantum.Intrinsic",
 ];
 
+struct IndentMaker<'a> {
+    contents: &'a str,
```

Comment:
I think you can just use a `SourceContents` (reference counted string, the same type that is stored in the `SourceMap`) here instead of having to futz with lifetimes.

Reply:
Avoided this by using contents without a need to annotate types, so it implicitly uses `SourceContents`.